### PR TITLE
Add support for horizontal/vertical flipping

### PR DESCRIPTION
**Pull request by dougg3**
There was a request in issue #144 for the ability to reflect/flip the video display. This commit adds a new -f flag for flipping, with three options you can supply:

- horiz (horizontal flip)
- vert (vertical fip)
- both (flip both horizontal and vertical, which would be the same as 180 degree rotation)

Tested on my Raspberry Pi. Only implemented in the rpi renderer for now.

### CODE REVIEW

**Comment by dougg3** [comment]
I updated this pull request to also implement flipping and rotation in the gstreamer renderer.

File: renderers/video_renderer.h [hunk 1]
```diff
@@ -71,9 +78,9 @@ typedef struct video_renderer_s {
     video_renderer_type_t type;
 } video_renderer_t;
 
-video_renderer_t *video_renderer_dummy_init(logger_t *logger, background_mode_t background_mode, bool low_latency, int rotation);
-video_renderer_t *video_renderer_rpi_init(logger_t *logger, background_mode_t background_mode, bool low_latency, int rotation);
-video_renderer_t *video_renderer_gstreamer_init(logger_t *logger, background_mode_t background_mode, bool low_latency, int rotation);
+video_renderer_t *video_renderer_dummy_init(logger_t *logger, background_mode_t background_mode, bool low_latency, int rotation, flip_mode_t flip);
```
Comment (pallas) [inline renderers/video_renderer.h:81]: Future TODO: it'd be nice if all these features got put into a configuration object and only that was passed to init.

Comment (pallas) [reply renderers/video_renderer.h:81]: Obviously, no need to do this now.

**Review by pallas — APPROVED** [review]
Overall looks good to me. Thanks for implementing this!

File: renderers/video_renderer.h (hunk 1 above)
Comment (dougg3) [reply renderers/video_renderer.h:81]: I totally agree, that would also make it easier as new capabilities are added!

Comment (FD-) [reply renderers/video_renderer.h:81]: +1 👍

**Comment by FD-** [comment]
Thanks also from my side! Btw @pallas would you be interested in getting write access to the repo? I sent you an email a while back regarding this, but you never responded 😕

**Comment by pallas** [comment]
@FD- I replied to you, maybe it went to spam though? Happy to take on write-access for reviews like this.

**Pull request merged**